pkg/types/ovirt/machinepool.go

```diff
@@ -3,11 +3,66 @@ package ovirt
 // MachinePool stores the configuration for a machine pool installed
 // on ovirt.
 type MachinePool struct {
+	// InstanceTypeID defines the VM instance type and overrides
+	// the hardware parameters of the created VM, including cpu and memory.
+	// If InstanceTypeID is passed, all memory and cpu variables will be ignored.
+	InstanceTypeID string `json:"instanceTypeID,omitempty"`
+
+	// CPU defines the VM CPU.
+	CPU *CPU `json:"cpu,omitempty"`
+
+	// MemoryMB is the size of a VM's memory in MiBs.
+	MemoryMB int32 `json:"memoryMB,omitempty"`
+
+	// OSDisk is the the root disk of the node.
+	OSDisk *Disk `json:"osDisk,omitempty"`
+
+	// VMType defines the workload type the instance will
+	// be used for and this effects the instance parameters.
+	// One of "desktop, server, high_performance"
+	VMType string `json:"type,omitempty"`
+}
+
+// CPU defines the VM cpu, made of (Sockets * Cores * Threads)
```

Comment:
nit: missing period

Reply:
```suggestion
// CPU defines the VM's cpu, which is made of (Sockets * Cores * Threads)
```

The sockets and cores are defined in this struct, while there is no mention of the threads?

Reply:
I'm hiding threads for now, I don't want to expose this at the moment. I removed it from the comments as well.